source/Plugins.cpp

```diff
@@ -164,3 +180,36 @@ void Plugins::TogglePlugin(const string &name)
 	auto *plugin = plugins.Get(name);
 	plugin->currentState = !plugin->currentState;
 }
+
+
+
+future<void> Plugins::Install(const PluginInstallData &installData)
+{
+	oldNetworkActivity = true;
+	return async(launch::async, [installData]() noexcept -> void
+		{
+			++currentBackgroundActivity;
+
+			bool success = PluginHelper::Download(installData.url.c_str(),
```

Comment:
All those c_str calls are unnecessary now that the functions take a std::string

Reply:
removed them